### PR TITLE
Updates - github actions

### PR DESCRIPTION
**Pull request by GernotMaier**
Several of the action flows are outdated:

- general update of version names 
- replace conda installation 'by hand' in CI-docs.yml by using mamba-org/setup-micromamba
- replace mamba-org/provision-with-micromamba by mamba-org/setup-micromamba (as indicated [here](https://github.com/marketplace/actions/provision-with-micromamba)

Note that it is interesting that there is no `:latest` flag for github actions and that one has to manually update the versions. There is a github action to update github action ([here](https://github.com/marketplace/actions/github-actions-version-updater)), but let's keep it simple - github is provided a deprecated warning:

Screenshot 2023-08-11 at 15 27 56

### CODE REVIEW

File: .github/workflows/CI-unittests.yml [hunk 1]
```diff
@@ -68,9 +68,21 @@ jobs:
...
+        if: matrix.install-method == 'mamba'
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        run: |
+          # setup correct python version
+          sed -i -e "s/- python=.*/- python=$PYTHON_VERSION/g" environment.yml
+
+      - name: Install mamba dependencies
+        if: matrix.install-method == 'mamba'
+        uses: mamba-org/setup-micromamba@v1
```
Comment (VictorBarbosaMartins) [inline .github/workflows/CI-unittests.yml:81]: In case we update the version to 1.4.3, please update it here too

**Review by VictorBarbosaMartins — APPROVED** [review]
Hi @GernotMaier. Thanks for this update. 
I am not the expert on github workflows but I tried my best. I wrote a couple of questions/suggestions, it is up to you to consider them or go on. I believe we will see the effect of this PR, especially on the docs, after we merge it.

I am already approving but let's at least wait for the tests to complete (I merged main into the branch).

File: .github/workflows/CI-docs.yml [hunk 2]
```diff
@@ -14,27 +14,17 @@ jobs:
 
     steps:
       - name: checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Set up Python 3.10.4
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply .github/workflows/CI-docs.yml:20]: Yes, unfortunately we have to. This is why we introduced the bot to check for version on certain time intervals and update it.

File: .github/workflows/CI-docs.yml [hunk 3]
```diff
@@ -14,27 +14,17 @@ jobs:
...
-          source $CONDA/etc/profile.d/conda.sh
-          conda config --set always_yes yes --set changeps1 no
-          conda update -q conda
-          sed -i -e "s/- python=3.9/- python=$PYTHON_VERSION/g" environment.yml
-          conda env create -f environment.yml
-          conda activate simtools-dev
-          echo 'source $CONDA/etc/profile.d/conda.sh' >> ~/.bash_profile
-          echo 'conda activate simtools-dev' >> ~/.bash_profile
-          echo 'pip install -e .' >> ~/.bash_profile
           python --version
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply .github/workflows/CI-docs.yml:26]: Not really, but it has saved me quite a few times while debugging. I think this is is a easy test that the installation went well.

File: .github/workflows/CI-docs.yml [hunk 4]
```diff
@@ -14,27 +14,17 @@ jobs:
 
     steps:
       - name: checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
       - name: Set up Python 3.10.4
```
[the remark this answers is not in the document]
Comment (GernotMaier) [reply .github/workflows/CI-docs.yml:19]: This is for the docs, I think here the python version is not so relevant (unless there is a python / sphinx dependencies). I think at this point we want things working with 3.9-3.11, so this should be goo.d

**Pull request merged**